pkg/istio-agent/xds_proxy.go

```diff
@@ -370,24 +414,25 @@ func (p *XdsProxy) handleUpstreamResponse(con *ProxyConnection) {
 			// TODO: separate upstream response handling from requests sending, which are both time costly
 			proxyLog.Debugf("response for type url %s", resp.TypeUrl)
 			metrics.XdsProxyResponses.Increment()
-			switch resp.TypeUrl {
-			case v3.NameTableType:
-				// intercept. This is for the dns server
-				if p.localDNSServer != nil && len(resp.Resources) > 0 {
-					var nt nds.NameTable
-					// TODO we should probably send ACK and not update nametable here
-					if err := ptypes.UnmarshalAny(resp.Resources[0], &nt); err != nil {
-						log.Errorf("failed to unmarshall name table: %v", err)
+			if h, f := p.handlers[resp.TypeUrl]; f {
+				err := h(resp)
+				var errorResp *google_rpc.Status
+				if err != nil {
+					errorResp = &google_rpc.Status{
+						Code:    13,
```

Comment:
My understanding was that Code is using the HTTP error codes - with some extensions. 
If there is a different namespace or table - it probably has some table, at least add a comment where the 13 is coming from.